### PR TITLE
qt-percona-server: revision bump (percona-server 8.0.22-13)

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This should fix the error seen in various CI runs (e.g. #73162, #71191)
```
==> brew linkage --test qt-percona-server
==> FAILED
Broken dependencies:
  /opt/homebrew/opt/percona-server/lib/libcrypto.1.1.dylib (percona-server)
  /opt/homebrew/opt/percona-server/lib/libssl.1.1.dylib (percona-server)
```
This was caused by `qt-percona-server` having been built before the `percona-server` version bump (#72505), but the PR where `qt-percona-server` was built wasn't merged until after the `percona-server` version bump.

As an aside, why is `percona-server` shipping its own SSL libraries?

### CODE REVIEW

**Comment by carlocab** [comment]
Those libraries shouldn't be there:

https://github.com/Homebrew/homebrew-core/blob/71ccd48f3217276bf2634b9698373d34ffb38607/Formula/percona-server.rb#L97-L103

**Comment by carlocab** [comment]
```
==> make sub-mysql
make: *** No rule to make target `sub-mysql'.  Stop.
```
CC @paperchalice

**Comment by paperchalice** [comment]
try to change (this line)[https://github.com/Homebrew/homebrew-core/blob/71ccd48f3217276bf2634b9698373d34ffb38607/Formula/qt-percona-server.rb#L25] to `system "qmake", "--", "MYSQL_INCDIR=#{Formula["percona-server"].include}"`,
from https://doc.qt.io/qt-5/sql-driver.html#qmysql

**Comment by Bo98** [comment]
> As an aside, why is `percona-server` shipping its own SSL libraries?

Because MySQL's (and by extension Percona Server's) build system only considers the package manager case on Linux. The Windows and macOS builds are very much designed to be the way they want to ship MySQL binaries, with no flexibility like on Linux.

**Comment by carlocab** [comment]
I guess I'm more confused about what those libraries are still doing there, given https://github.com/Homebrew/homebrew-core/pull/73166#issuecomment-799027146.

**Comment by carlocab** [comment]
Still the same error on Mojave, but build succeeds on ARM. Very weird. @paperchalice

**Comment by Bo98** [comment]
They're not - as you say, `qt-percona-server` was built before the `percona-server` version bump where I removed those libraries.

**Comment by carlocab** [comment]
Ah, I didn't notice those lines were new. That makes sense.

**Comment by Bo98** [comment]
As for the error I suggest:

* Using `cmake` instead of `qmake` (the former is preferred in Qt 6).
* Adding `depends_on "pkg-config" => :build` (if it can't find `percona-server` on its own).

**Comment by Bo98** [comment]
(Or, alternatively `qt-cmake`).

**Comment by paperchalice** [comment]
> As for the error I suggest:
> 
> * Using `cmake` instead of `qmake` (the former is preferred in Qt 6).
> * Adding `depends_on "pkg-config" => :build` (if it can't find `percona-server` on its own).
+1, the `qmake` project files are removed in Qt6.1, related CMake variable is `QT_FEATURE_sql_<database>`
https://github.com/qt/qtbase/blob/6.0/src/plugins/sqldrivers/CMakeLists.txt

**Comment by paperchalice** [comment]
The install steps will be:
```
system "cmake", ".", "-DCMAKE_STAGING_PREFIX=#{prefix}", *std_cmake_args
cd "mysql" do
  system "make"
  system "make", "install
end
```

**Comment by Bo98** [comment]
It would be nice if we could also disable anything not-MySQL if possible, seems like this is currently set up to build everything available.

**Comment by paperchalice** [comment]
> It would be nice if we could also disable anything not-MySQL. Seems like this is currently set up to build everything available.

updated!

**Comment by carlocab** [comment]
> The install steps will be:
> 
> ```
> args = std_cmake_args +  %W[
>   -DCMAKE_STAGING_PREFIX=#{prefix}
> 
>   -DFEATURE_sql_ibase=OFF
>   -DFEATURE_sql_mysql=ON
>   -DFEATURE_sql_oci=OFF
>   -DFEATURE_sql_odbc=OFF
>   -DFEATURE_sql_psql=OFF
>   -DFEATURE_sql_sqlite=OFF
> ]
> system "cmake", ".", *args
> system "make"
> system "make", "install"
> ```

Will this install in the right location? At the moment the formula installs into `share/"qt"` manually.

**Comment by paperchalice** [comment]
> > The install steps will be:
> > ```
> > args = std_cmake_args +  %W[
> >   -DCMAKE_STAGING_PREFIX=#{prefix}
> > 
> >   -DFEATURE_sql_ibase=OFF
> >   -DFEATURE_sql_mysql=ON
> >   -DFEATURE_sql_oci=OFF
> >   -DFEATURE_sql_odbc=OFF
> >   -DFEATURE_sql_psql=OFF
> >   -DFEATURE_sql_sqlite=OFF
> > ]
> > system "cmake", ".", *args
> > system "make"
> > system "make", "install"
> > ```
> 
> Will this install in the right location? At the moment the formula installs into `share/"qt"` manually.

I ran these steps in my machine, they are installed into the correct place.

File: Formula/qt-percona-server.rb [hunk 1]
```diff
@@ -21,10 +23,21 @@ class QtPerconaServer < Formula
...
     cd "src/plugins/sqldrivers" do
-      system "qmake"
-      system "make", "sub-mysql"
-      (share/"qt").install "plugins/"
+      system "cmake", "-S", ".", "-B", "build", *args
+      system "cmake", "--build", "build"
+      system "cmake", "--install", "build"
     end
     Pathname.glob(share/"qt/plugins/sqldrivers/#{shared_library("*")}") do |plugin|
       system "strip", "-S", "-x", plugin
```
Comment (paperchalice) [inline Formula/qt-percona-server.rb:43]: This step now is redundant.

**Comment by carlocab** [comment]
Hmmm... still can't find `percona-server`.

**Comment by paperchalice** [comment]
After inspecting the `FindMySQL.cmake` in qt source tree, it only tries to use `pkg-config` to find `libmysql mysql mysqlclient libmariadb mariadb`.
so add `-DMySQL_LIBRARIES=#{Formula["percona-server"].opt_lib}/libperconaserverclient.dylib` to cmake args

**Comment by carlocab** [comment]
`pkg-config` is already a build dependency. I'll try your other suggestion.

**Comment by Bo98** [comment]
After this is merged, the changes should ideally be ported over to the other Qt SQL drivers too.

**Comment by carlocab** [comment]
> After this is merged, the changes should ideally be ported over to the other Qt SQL drivers too.

Do you want to take care of this, @paperchalice?

**Comment by paperchalice** [comment]
I will do it.

**Pull request closed without merging**